### PR TITLE
add impala engineconn-plugin

### PR DESCRIPTION
**Pull request by ws00428637**
### What is the purpose of the change

EngineConn-Plugin for Impala

### Related issues/PRs

Related issues: #4180

### Brief change log

- Define an executor base on Impala thrift client;
- Fix some code styles along with Trino engineconn-plugin;

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [x] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Review by jacktao007 — DISMISSED** [review]
LGTM.

File: linkis-engineconn-plugins/impala/src/main/resources/log4j2.xml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<configuration status="error" monitorInterval="30">
+    <appenders>
+        <Console name="Console" target="SYSTEM_OUT">
+            <ThresholdFilter level="INFO" onMatch="ACCEPT" onMismatch="DENY"/>
+            <PatternLayout pattern="%d{yyyy-MM-dd HH:mm:ss.SSS} %-5level [%t] %logger{36} %L %M - %msg%xEx%n"/>
```
Comment (peacewong) [inline linkis-engineconn-plugins/impala/src/main/resources/log4j2.xml:23]: log4j2 can refer to the configuration of hive ecp

File: linkis-engineconn-plugins/impala/src/main/scala/org/apache/linkis/engineplugin/impala/executor/ImpalaEngineConnExecutor.scala [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ArrayBuffer
+
+class ImpalaEngineConnExecutor(override val outputPrintLimit: Int, val id: Int)
+    extends ConcurrentComputationExecutor(outputPrintLimit) {
+
+  private val executorLabels: util.List[Label[_]] = new util.ArrayList[Label[_]](2)
+
+  private val impalaClient: ImpalaClient = {
+    val servers = IMPALA_SERVERS.getValue.split(',')
```
Comment (peacewong) [inline linkis-engineconn-plugins/impala/src/main/scala/org/apache/linkis/engineplugin/impala/executor/ImpalaEngineConnExecutor.scala:92]: Can addresses support task adjustments?

File: pom.xml [hunk 3]
```diff
@@ -110,6 +110,9 @@
     <io_file.version>1.0</io_file.version>
     <jdbc.version>4</jdbc.version>
     <trino.version>371</trino.version>
+    <impala.version>3.4.0.7.2.15.0-147</impala.version>
+    <doris-connector.artifact>flink-doris-connector-1.15</doris-connector.artifact>
```
Comment (peacewong) [inline pom.xml:114]: doris can remove

**Comment by casionone** [comment]
The unit test failed at compile time, please see why.
And the code seems to conflict.

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**